src/gtest/test_checkblock.cpp

```diff
@@ -67,27 +67,99 @@ TEST(CheckBlock, BlockSproutRejectsBadVersion) {
 }
 
 
-TEST(ContextualCheckBlock, BadCoinbaseHeight) {
-    SelectParams(CBaseChainParams::MAIN);
+class ContextualCheckBlockTest : public ::testing::Test {
+protected:
+    virtual void SetUp() {
+        SelectParams(CBaseChainParams::MAIN);
+    }
+
+    virtual void TearDown() {
+        // Revert to test default. No-op on mainnet params.
+        UpdateNetworkUpgradeParameters(Consensus::UPGRADE_OVERWINTER, Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT);
+        UpdateNetworkUpgradeParameters(Consensus::UPGRADE_SAPLING, Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT);
```

Comment:
Not that it should materially affect anything, but I'd prefer to revert these in the opposite order, so the global state remains consistent (we shouldn't have Sapling active and Overwinter not).